engine/api/pipeline_test.go

```diff
@@ -264,3 +264,96 @@ func Test_runPipelineWithLastParentHandler(t *testing.T) {
 	assert.Equal(t, pb.Environment.ID, pb1.Trigger.ParentPipelineBuild.Environment.ID)
 
 }
+
+func Test_deletePipelineHandler(t *testing.T) {
+	api, db, router := newTestAPI(t, bootstrap.InitiliazeDB)
+
+	//1. Create admin user
+	u, pass := assets.InsertAdminUser(api.mustDB())
+	//2. Create project
+	proj := assets.InsertTestProject(t, db, api.Cache, sdk.RandomString(10), sdk.RandomString(10), u)
+	test.NotNil(t, proj)
+
+	//3. Create Pipeline
+	pipelineKey := sdk.RandomString(10)
+	pip := &sdk.Pipeline{
+		Name:       pipelineKey,
+		Type:       sdk.BuildPipeline,
+		ProjectKey: proj.Key,
+		ProjectID:  proj.ID,
+	}
+	test.NoError(t, pipeline.InsertPipeline(api.mustDB(), api.Cache, proj, pip, nil))
+
+	vars := map[string]string{
+		"key":             proj.Key,
+		"permPipelineKey": pip.Name,
+	}
+	uri := router.GetRoute("DELETE", api.deletePipelineHandler, vars)
+	test.NotEmpty(t, uri)
+
+	req, err := http.NewRequest("DELETE", uri, nil)
+	if err != nil {
+		t.FailNow()
+		return
+	}
+	assets.AuthentifyRequest(t, req, u, pass)
+
+	//8. Do the request
+	w := httptest.NewRecorder()
+	router.Mux.ServeHTTP(w, req)
+
+	assert.Equal(t, 204, w.Code)
+
+	pip2, err := pipeline.LoadPipeline(db, proj.Key, pip.Name, false)
+	assert.Nil(t, pip2)
+	assert.Error(t, err)
+}
+
+func Test_deletePipelineHandlerShouldReturnError(t *testing.T) {
```

Comment:
That cover the use case of old workflow.
Need a test with a real workflow

Reply:
done